lm_eval/api/task.py

```diff
@@ -85,7 +101,7 @@ class TaskConfig(dict):
     filter_list: Union[str, list] = None
     should_decontaminate: bool = False
     doc_to_decontamination_query: str = None
-
+    weight_by_size: bool = False
```

Comment:
do we still need `weight_by_size` in `TaskConfig`?

Reply:
Yes, I think so, even if it's set in a group config that would then be set to each task.

Reply:
How does this work if someone calls both `mmlu` and `multimedqa` (supposing MMLU weights by size, but multimedqa does not)?

Won't the two compete for what `weight_by_size` should be set to for the mmlu subtasks they share?

Reply:
We would write `weight_by_size: true` in the config for `mmlu` which would make all tasks be weighted by its sample size. No need to do anything to `multimedqa` since `weight_by_size` is False.

I've remove operations related to `weight_by_size` in this PR and have the other PR continue it instead.

Reply:
I think this line still has `weight_by_size` in it